src/services/utilities.ts

```diff
@@ -924,11 +924,23 @@ namespace ts {
         }
     }
 
+    export function isPossiblyTypeArgumentPosition(token: Node, sourceFile: SourceFile, checker: TypeChecker): boolean {
+        const info = getPossibleTypeArgumentsInfo(token, sourceFile);
+        return info !== undefined && getPossibleGenericSignatures(info.called, info.nTypeArguments, checker).length !== 0;
+    }
+
+    export function getPossibleGenericSignatures(called: Expression, typeArgumentCount: number, checker: TypeChecker): ReadonlyArray<Signature> {
+        const type = checker.getTypeAtLocation(called)!; // TODO: GH#18217
+        const signatures = isNewExpression(called.parent) ? type.getConstructSignatures() : type.getCallSignatures();

```

Comment:
the whole purpose of this function was to do syntactic scan on tokens since the tree cant be guaranteed. Eg. what if someone was writing ```new Type<Type<```